conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosIndexer.scala

```diff
@@ -371,10 +371,56 @@ class TezosIndexer(
         delegateCheckpoints <- processAccountsForBlocks(accountUpdates, rollsData) // should this fail, we still recover data from the checkpoint
         _ <- processBakersForBlocks(delegateCheckpoints) // same as above
         _ <- processBlocksForGovernance(blocks, votingData)
+        _ <- updateBakers(blocks)
       } yield results.size
 
     }
 
+    /** Updates bakers in the DB */
+    def updateBakers(blocks: List[Block]): Future[Option[Unit]] = {
+      import cats.implicits._
+      logger.info("Updating Bakers table")
+      blocks
+        .find(_.data.header.level % lorreConf.blockRightsFetching.cycleSize == 1)
+        .map { block =>
+          val bakingRights = db.run(TezosDb.getBakingRightsForLevel(block.data.header.level))
+          val endorsingRights = db.run(TezosDb.getEndorsingRightsForLevel(block.data.header.level))
+          val bakersFromDb = db.run(TezosDb.getBakers)
```

Comment:
I think it will be heavier the way you described because in database we would need to use complex query which checks each one of the bakers against a set of values.

Reply:
Also we have only ~600 bakers in the mainnet, so this way should be more optimal.

Reply:
Ok, let's go with it